### PR TITLE
Feature 171 Update form versions

### LINKED ISSUE #171: Use MetaformVersions update method
As a user I'd like to be able to correctly update versions instead of deleting old and creating new.

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #171 
Also adds confirmation dialog for when publishing draft form.
Also minor bugfix related to navigating to draft editor when draft has different slug than corresponding Metaform

### CODE REVIEW

File: src/components/editor/editor-screen-table.tsx [hunk 1]
```diff
@@ -103,20 +103,26 @@ const EditorScreenTable: FC<Props> = ({
    */
   const buildVersionRows = (metaform: Metaform): MetaformVersionRow[] => {
     const versions = metaformVersions.filter(x => x.data.id as unknown === metaform.id);
+    versions.sort((a, b) => {
+      return a.modifiedAt!.getTime() > b.modifiedAt!.getTime() ? -1 : 1;
+    });
```
Comment (Chi-Chu319) [inline src/components/editor/editor-screen-table.tsx:108]:
```suggestion
    versions.sort((a, b) => (a.modifiedAt!.getTime() > b.modifiedAt!.getTime() ? -1 : 1));
```

File: src/components/editor/editor-screen-table.tsx [hunk 2]
```diff
@@ -103,20 +103,26 @@ const EditorScreenTable: FC<Props> = ({
    */
   const buildVersionRows = (metaform: Metaform): MetaformVersionRow[] => {
     const versions = metaformVersions.filter(x => x.data.id as unknown === metaform.id);
+    versions.sort((a, b) => {
+      return a.modifiedAt!.getTime() > b.modifiedAt!.getTime() ? -1 : 1;
```
Comment (Chi-Chu319) [inline src/components/editor/editor-screen-table.tsx:107]: Dangerous to use null assertion, maybe filter first?

File: src/components/screens/editor/editor-screen.tsx [hunk 3]
```diff
@@ -141,7 +142,7 @@ const EditorScreen: React.FC = () => {
         }
       });
 
-      navigate(`${currentPath}/${metaformToEdit.slug}/${newMetaformVersion.id}`);
+      return navigate(`${currentPath}/${metaformToEdit.slug}/${newMetaformVersion.id}`);
```
Comment (Chi-Chu319) [inline src/components/screens/editor/editor-screen.tsx:145]: Is it necessary to return?

File: src/localization/en.json [hunk 4]
```diff
@@ -189,6 +189,10 @@
   "draftEditorScreen": {
     "preview": "Preview",
     "publish": "Publish",
+    "publishDialog": {
+      "title": "Are you sure?",
```
Comment (Chi-Chu319) [inline src/localization/en.json:193]: Could be changed to just Publish version (as well as in Finnish localization)

**Review by Chi-Chu319 — APPROVED** [review]

File: src/components/editor/editor-screen-table.tsx (hunk 2 above)
Comment (villejuutila) [reply src/components/editor/editor-screen-table.tsx:107]: Good idea.

**Pull request merged**